### PR TITLE
fix detail view of ProtVista component and initiate protractor tests

### PR DESCRIPTION
**Pull request by jayjayjpg**
@emepyc @4ndr01d3 @ljgarcia @wilzbach Please review and merge to stable if this looks suitable: dev.biojs.io

### CODE REVIEW

File: angular/component-table/component-table-view.html [hunk 1]
```diff
@@ -9,14 +9,14 @@
     <col width="100px" />
     <thead>
     <tr>
-       <th>  
+       <th>
         <th class="link" ng-click="setOrderProp('name')" ng-class="[{active: orderProp == 'name'},'listview-pointer']" > <nobr><i class="fa fa-bookmark-o"></i> Component</nobr>
         <!-- <th>Summary -->
 <!--         <th ng-click="setOrderProp('version')" ng-class="[{active: orderProp == 'version'},'listview-pointer']">Version
- -->    
+ -->
```
Comment (ljgarcia) [inline angular/component-table/component-table-view.html]: If not needed, those comments could be removed so the code would be cleaner.

File: angular/component-table/component-table-view.html [hunk 2]
```diff
@@ -9,14 +9,14 @@
...
         <th class="link" ng-click="setOrderProp('stars')" ng-class="[{active: orderProp == 'stars'},'listview-pointer']"><nobr><i class="fa fa-star"></i> Stars</nobr>
-        <th class="link" ng-click="setOrderProp('downloads')" ng-class="[{active: orderProp == 'downloads'},'listview-pointer']"><nobr><i class="fa fa-download"></i> Downloads</nobr> 
-       
+        <th class="link" ng-click="setOrderProp('downloads')" ng-class="[{active: orderProp == 'downloads'},'listview-pointer']"><nobr><i class="fa fa-download"></i> Downloads</nobr>
+
```
Comment (ljgarcia) [inline angular/component-table/component-table-view.html:16]: Other than the removed spaces, I fail to see the difference between the previous and the new lines. Was the purpose here to clean out spaces?

File: angular/component-table/component-table-view.html [hunk 3]
```diff
@@ -45,18 +45,18 @@
             <div style="float: left; display: inline-block; margin-right: 10px">
 				<h2 class="link"  style="margin-top: 0; white-space: nowrap; font-size:18px" class="ng-binding"> {{ component.name }} <span style="font-size:10px">{{component.version}} </span> <br></h2>
 				<div style="font-size:14px; margin-top: -7px;padding-bottom: 5px">{{ component.author.name }}</div>
-                <p class="ng-binding"> {{component.description | limitTo: 120 }}{{ component.description.length > 50 ? '...' : '' }}</i></p> 
+                <p class="ng-binding"> {{component.description | limitTo: 120 }}{{ component.description.length > 50 ? '...' : '' }}</i></p>
```
Comment (ljgarcia) [inline angular/component-table/component-table-view.html:45]: See previous comment

File: angular/component-model/component-model.js [hunk 4]
```diff
@@ -234,6 +234,8 @@ angular.module('registry').service("Component", function ($http, $window, $sce,
               if(all.length === 0){
                 Object.keys(resp).forEach(function(key) {
                   all.push(resp[key]);
+                 // console.log("resp: " + JSON.stringify(resp));
+                  // console.log("resp key: " + JSON.stringify(resp[key]));
```
Comment (ljgarcia) [inline angular/component-model/component-model.js]: If those lines are not needed anymore, they could be removed.

File: protractor/spec.js [hunk 5]
```diff
@@ -19,17 +19,19 @@ describe('biojs registry', function() {
   var hiddenCount = element(by.binding("hiddenCount"));
   var table = element.all(by.repeater("component in components | filter:filterFn"));
   var seqTag = element(by.name("sequence"));
+  // httpBackend.when("GET", "http://dev.workmen.biojs.net/all?short=1").respond(myMock);
```
Comment (ljgarcia) [inline protractor/spec.js]: We could remove this commented line.

File: angular/component-model/component-model.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 var ComponentService = angular.module('registry').service("Component", function ($http, $window, $sce, $q) {
 
     function preProcessComponent(component) {
+        console.log("new revision");
```
Comment (ljgarcia) [inline angular/component-model/component-model.js]: Do we really need this console.log here? If just for debugging purposes, better remove it from the final commit.

File: angular/component-table/component-table-view.html (hunk 2 above)
Comment (jayjayjpg) [inline angular/component-table/component-table-view.html:16]: Sometimes the git client will give error messages on commit when there is plain whitespace left in empty lines of the submitted code. I assume @ayusharma wanted to get rid of those here, which is a good thing!

File: angular/component-table/component-table-view.html (hunk 1 above)
Comment (jayjayjpg) [inline angular/component-table/component-table-view.html]: Thanks that's true, I'll update this!

File: angular/component-model/component-model.js (hunk 4 above)
Comment (jayjayjpg) [inline angular/component-model/component-model.js]: Good point, thanks!

File: angular/component-table/component-table-view.html (hunk 2 above)
Comment (ljgarcia) [inline angular/component-table/component-table-view.html:16]: Oh, so it is a styling issue, sorry, I did not know about it. Indeed, better to get rid of those then.

**Comment by jayjayjpg** [comment]
@ljgarcia @emepyc @4ndr01d3 Hey everyone, I'd like to merge this tonight to fix the bug with the display of the ProtVista component in our registry. Please object if you find any issues in this PR that speak against a merge

**Comment by ljgarcia** [comment]
Hi Jessica, although I was not able to perform an in-depth code review, the pages are working fine for me. As some changes are related to styling and other visual stuff, for the future, I would suggest to include an example to check for reviewers (particularly for those of us not well familiar with the code).

**Comment by jayjayjpg** [comment]
@ljgarcia Thanks for the feedback! The code can by the way be checked without any local installation anytime on http://dev.biojs.io where the gh-pages branch is deployed (see also in the [README](https://github.com/biojs/registry-ui#dev-instance)). I already merged all the changes from this PR to this gh-pages branch so anyone is free to test it there

EDIT: Ah sorry, I now get it! I'll attach a couple of screenshots next time, that's a good point especially for the visual changes!

**Pull request merged**